app/consolidator/services/ConsolidatorService.scala

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package consolidator.services
+
+import java.io.{ BufferedWriter, File, FileWriter }
+import java.time.Instant
+
+import cats.effect.Resource.fromAutoCloseable
+import cats.effect.{ ContextShift, IO, Resource }
+import cats.implicits._
+import collector.repositories.FormRepository
+import consolidator.IOUtils
+import consolidator.repositories.{ ConsolidatorJobData, ConsolidatorJobDataRepository }
+import consolidator.services.ConsolidatorService.ProcessFormsResult
+import javax.inject.{ Inject, Singleton }
+import org.slf4j.{ Logger, LoggerFactory }
+import play.api.Configuration
+import reactivemongo.bson.BSONObjectID
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+@Singleton
+class ConsolidatorService @Inject()(
+  formRepository: FormRepository,
+  consolidatorJobDataRepository: ConsolidatorJobDataRepository,
+  config: Configuration
+)(implicit ec: ExecutionContext)
+    extends IOUtils {
+
+  private val logger: Logger = LoggerFactory.getLogger(getClass)
+  implicit val contextShift: ContextShift[IO] = IO.contextShift(ec)
+  private val batchSize = config.underlying.getInt("consolidator-job-config.batchSize")
+
```

Comment:
- Create a output temp file in write mode
- Fetch the last object id from the previous run, from the consolidator_job_datas collection
- Get the forms metadata (count and max-id), where object id is greater than last object id (from step 2)
- Fetch all the form data from the forms collection in batches, where object id is greater than last object id and less than max-id (from step 2 & 3)
- Transform each form to a JSON line
- Add the generate JSON line to the output file (step 1)
- Return the generated output file name (step1) or None if the file could not be generated due to no data